src/name.rs

```diff
@@ -0,0 +1,28 @@
+//! Support for associating type name information with a [`Message`].
+
+use crate::Message;
+use alloc::{format, string::String};
+
+/// Associate a type name with a [`Message`] type.
+pub trait Name: Message {
```

Comment:
Can this just live in `prost_types`, why does it need to live in the core crate?

Reply:
I was hoping it could eventually be autogenerated by `prost-build`, at least optionally. We're hand annotating them right now and it's getting quite cumbersome.

It seems like `prost-build` currently only refers to types/traits in `prost`. If it were optional and it could refer to types/traits in `prost-types`, I'd be fine with the `Name` trait living there.

Reply:
I see your point, do we just want to go ahead and do the work to also add the prost-build parts? The next release will be breaking and I would be ok if we add something like that.